cpp/src/arrow/array/array_nested.h

```diff
@@ -127,11 +127,15 @@ class ARROW_EXPORT ListArray : public BaseListArray<ListType> {
   /// allocated because of null values
   static Result<std::shared_ptr<ListArray>> FromArrays(
       const Array& offsets, const Array& values,
-      MemoryPool* pool = default_memory_pool());
+      MemoryPool* pool = default_memory_pool(),
+      int64_t null_count = -1,
+      std::shared_ptr<Buffer> null_bitmap = NULLPTR);
```

Comment:
Good point. I noticed the ordering part, but thought placing it after memory pool would lead to less API breakage since the two added have defaults. Maybe it's better to have parameter order consistent. 👍